systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectIsolatedST.java

```diff
@@ -1318,6 +1322,71 @@ void testKafkaConnectWithScramShaAuthenticationRolledAfterPasswordChanged(Extens
         KafkaConnectUtils.waitUntilKafkaConnectRestApiIsAvailable(namespaceName, kafkaConnectPodNameAfterRU);
     }
 
+    @ParallelNamespaceTest
+    @Tag(INTERNAL_CLIENTS_USED)
+    // When adding unexpected config option (tasks.max) - connectors are restarted on every reconciliation
+    void testKafkaConnectRestartingConnector(ExtensionContext extensionContext) throws InterruptedException {
+        TestStorage storage = new TestStorage(extensionContext);
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(storage.getClusterName(), 3).build());
+        resourceManager.createResource(extensionContext, KafkaConnectTemplates.kafkaConnect(extensionContext, storage.getClusterName(), 1, false)
+            .editMetadata()
+                .addToAnnotations(Annotations.STRIMZI_IO_USE_CONNECTOR_RESOURCES, "true")
+            .endMetadata()
+            .editOrNewSpec()
+                .withBootstrapServers(KafkaResources.tlsBootstrapAddress(storage.getClusterName()))
+                .addToConfig("key.converter.schemas.enable", false)
+                .addToConfig("value.converter.schemas.enable", false)
+                .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .withNewInlineLogging()
+                    .addToLoggers("connect.root.logger.level", "INFO")
+                .endInlineLogging()
+            .endSpec()
+            .build());
+
+        resourceManager.createResource(extensionContext, KafkaConnectorTemplates.kafkaConnector(storage.getClusterName())
+            .editSpec()
+                .withTasksMax(2)
+                .withClassName("org.apache.kafka.connect.file.FileStreamSinkConnector")
+                .addToConfig("topics", storage.getTopicName())
+                .addToConfig("file", Constants.DEFAULT_SINK_FILE_PATH)
+            .endSpec()
+            .build());
+
+        String connectorName = storage.getClusterName();
+        String coExpectedLog = "KafkaConnect(" + storage.getNamespaceName() + "/" + connectorName + "): Configuration option \"tasks.max\" is forbidden and will be ignored";
```

Comment:
Its confusing if you call it _expected_ when it is expected in one case and unexpected in the other.